desc/objectives/utils.py

```diff
@@ -130,32 +141,51 @@ def factorize_linear_constraints(objective, constraint):  # noqa: C901
             )
         A = A[unfixed_rows][:, unfixed_idx]
         b = b[unfixed_rows]
+
     unfixed_idx = indices_idx
+    fixed_idx = np.delete(np.arange(xp.size), unfixed_idx)
+
+    # get x_scale if not provided
+    if x_scale == "auto":
+        if things is None:
+            things = objective.things
+        elif not isinstance(things, list):
+            things = [things]
+        x_scale = objective.x(*[things[things.index(t)] for t in objective.things])
+    D = np.where(np.abs(x_scale) < 1e1, 1, np.abs(x_scale))  # TODO: adjust threshold?
```

Comment:
I increased the threshold to `10` because otherwise some tests were failing. This basically sets it so that the scaling is not applied to the usual R, Z, Lambda, etc. coefficients and will only affect parameters like current, density, temperature, etc.